proposals/devcontainer-distribution.md

```diff
@@ -0,0 +1,71 @@
+| version | string | Version information of this collection list.|
+| features | array | The list of features that are contained in this collection. Same information as the metadata file for the feature.|
+| templates | array | The list of templates contained in this collection. Same information as the metadata file for the definition.|
+
+Its important to note the following about collections:
+
+- It is recomended that they are created automatically as part of the release process.
+- Ids would get the collection repository added to them. (e.g. for 'https://github.com/community/features' then 'community/features' to 'myfeature1' for a feature with that id)
+- The links to the release files would be added.
+- Date.
+- Version.
+
+## Templates specifics
+
+By their nature templates are single use since they are just a group of files that are copied over to an end users repository and used to kickstart a **dev container** configuration.
+
+Due to this the implementing application needs to ensure that the user is able to inspect all files included in the template and that no code is executed automatically until a specific action by a user demands it. 
+
+## Features specifics
+
+Since features contain code that can be executed multiple times by different users the security of the contents is important, for this we propose:
+
+- Any application that implements the [Dev container spec](../docs/specs/devcontainer-reference.md) must provide a way for the user to add a feature to their `devcontainer.json` file in a way that the code can be inspected before it executes the first time.
+- This action should create a **devcontainer.lock** file next to the main `devcontainer.json` file that contains the checksums of the particular release referenced.
+- Users can commit this file to their repository so that the application can berify the checksum each time it rebuilds the container, throwing an error if they not match.
+- If users want to update to a newer version they can either delete the **devcontainer.lock** file, delete the particular entry or reference a different version of it on their `devcontainer.json` file. If the version stored in both files doesn't match, the application should update the entry on the next build.
+- If during a first build the **devcontainer.lock** file does not exist, it should be created as part of the build process if and when it succeeds.
+
+The **devcontainer.lock** contains an array of:
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | The fully qualified id of the feature as defined [here](./devcontainer-features.md#devcontainerjson-properties).|
+| options | string | The same options object passed to the feature in `devcontainer.json`|
+| checksum | string | The checksum of the downloaded compressed release file.|
+
+It is important to note that, like in `devcontainer.json`  features can appear more than one time and are assumed to be in the same order.
```

Comment:
```suggestion
> **Note:** It is important to note that, like in `devcontainer.json`, features can appear more than one time and are assumed to be in the same order.
```

Reply:
What's a scenario where a feature would appear more than one time? And they're assumed to be in the same order as what?

Reply:
Reworded.